ipfx/ephys_data_set.py

```diff
@@ -41,36 +49,37 @@ def filtered_sweep_table(self,
                              ):
 
         st = self.sweep_table
-
         if current_clamp_only:
-            st = st[st[self.STIMULUS_UNITS].isin(self.current_clamp_units)]
+            mask = st[self.STIMULUS_UNITS].isin(self.current_clamp_units)
+            st = st[mask.astype(bool)]
```

Comment:
What issue is the `astype` fixing? Can we get a test for that?

Reply:
the `mask` is a Series of either dtype =bool or object. When mask's dtype = object and the filtered dataframe is empty, the header gets lost. Thus, the easy fix was to force mask's type to `bool`. Mostly likely there is a more robust solution to this problem but I have not found it. I will add a test for it as you suggested.

Reply:
I added the test `test_empty_table_has_header`
if you we do not use `astype(bool)` when filtering for `stimuli`, then the header is gone.